shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java

```diff
@@ -375,6 +375,31 @@ public void itDoesNotCrashWhenSplashScreenMetadataIsNotDefined() {
     assertNull(splashScreen);
   }
 
+  @Test
+  public void itDoesNotReleaseEnginewhenDetachFromFlutterEngine() {
+    FlutterActivityAndFragmentDelegate mockDelegate =
+        mock(FlutterActivityAndFragmentDelegate.class);
+    isDelegateAttached = true;
+    when(mockDelegate.isAttached()).thenAnswer(invocation -> isDelegateAttached);
+    doAnswer(invocation -> isDelegateAttached = false).when(mockDelegate).onDetach();
+    FlutterEngine mockEngine = mock(FlutterEngine.class);
+    FlutterEngineCache.getInstance().put("my_cached_engine", mockEngine);
+
+    Intent intent =
+        FlutterActivity.withCachedEngine("my_cached_engine").build(RuntimeEnvironment.application);
+    ActivityController<FlutterActivity> activityController =
+        Robolectric.buildActivity(FlutterActivity.class, intent);
+    FlutterActivity flutterActivity = activityController.get();
+
+    flutterActivity.setDelegate(mockDelegate);
+    flutterActivity.onStart();
+    flutterActivity.onResume();
+    flutterActivity.onPause();
+    flutterActivity.detachFromFlutterEngine();
+    assertNotNull(mockDelegate);
+    assertFalse(mockDelegate.isAttached());
```

Comment:
this works, but can we also add a test where `mockDelegate` is indeed `null`. This ensures that the test itself is working as one would expect.

Reply:
done.